pkg/reconciler/sequence/sequence.go

```diff
@@ -224,3 +232,87 @@ func (r *Reconciler) trackAndFetchChannel(ctx context.Context, seq *v1.Sequence,
 	}
 	return obj, err
 }
+
+func (r *Reconciler) removeUnwantedChannels(ctx context.Context, channelResourceInterface dynamic.ResourceInterface, seq *v1.Sequence, wanted []*eventingduckv1.Channelable) error {
+	channelObjRef := corev1.ObjectReference{
+		Kind:       seq.Spec.ChannelTemplate.Kind,
+		APIVersion: seq.Spec.ChannelTemplate.APIVersion,
+	}
+
+	l, err := r.channelableTracker.ListerFor(channelObjRef)
+	if err != nil {
+		logging.FromContext(ctx).Errorw("Error getting lister for Channel", zap.Any("channelRef", channelObjRef), zap.Error(err))
+		return err
+	}
+
+	exists, err := l.ByNamespace(seq.GetNamespace()).List(labels.Everything())
+	if err != nil {
+		logging.FromContext(ctx).Errorw("Error listing Channels", zap.Any("namespace", seq.Namespace), zap.Any("channelRef", channelObjRef), zap.Error(err))
+		return err
+	}
+
+	for _, c := range exists {
+		channelable, ok := c.(*eventingduckv1.Channelable)
+		if !ok {
+			logging.FromContext(ctx).Errorw("Failed to convert to Channelable Object", zap.Any("channel", c), zap.Error(err))
+			return fmt.Errorf("failed to convert to Channelable Object: %+v", c)
+		}
+
+		if !channelable.GetDeletionTimestamp().IsZero() ||
+			!metav1.IsControlledBy(channelable, seq) {
+			continue
+		}
+
+		used := false
+		for _, ch := range wanted {
+			if equality.Semantic.DeepDerivative(ch, channelable) {
+				used = true
+				break
+			}
+		}
+
+		if !used {
+			err = channelResourceInterface.Delete(ctx, channelable.Name, metav1.DeleteOptions{})
+			if err != nil {
+				logging.FromContext(ctx).Errorw("Failed to delete Channel", zap.Any("channel", channelable), zap.Error(err))
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *Reconciler) removeUnwantedSubscriptions(ctx context.Context, seq *v1.Sequence, wanted []*eventingduckv1.Channelable) error {
+	subs, err := r.subscriptionLister.Subscriptions(seq.Namespace).List(labels.Everything())
+	if err != nil {
+		logging.FromContext(ctx).Errorw("Error listing Subscriptions", zap.Any("namespace", seq.Namespace), zap.Error(err))
+		return err
+	}
+
+	for _, sub := range subs {
+		if !sub.GetDeletionTimestamp().IsZero() ||
+			!metav1.IsControlledBy(sub, seq) {
+			continue
+		}
+
+		used := false
+		for _, ch := range wanted {
+			if sub.Spec.Channel.APIVersion == ch.APIVersion &&
```

Comment:
same here, use the subscription name to compare

Reply:
Done at https://github.com/knative/eventing/pull/5718/commits/ad8aee54d965bf527d74fa76246d25b5d589a979